### PR TITLE
create user if missing if setting is turned on

### PR DESCRIPTION
**Pull request by i-just**
### Description
Adds `createUserIfMissing` setting (`false` by default). 

If it’s set to `true`, when a subscription is created, we check if the email address belongs to a user in Craft. If it doesn’t, a new (inactive) user is created.

### Related issues
#41

### CODE REVIEW

File: src/models/Settings.php [hunk 1]
```diff
@@ -68,6 +68,12 @@ class Settings extends Model
      */
     public ?string $defaultCancelUrl = null;
 
+    /**
+     * @var bool Whether creating a subscription for an email address that doesn't belong to any user,
+     * should create an inactive user.
+     */
```
Comment (nfourtythree) [inline src/models/Settings.php:75]:
```suggestion
     * @since 1.3.0
     */
```

File: src/services/Subscriptions.php [hunk 2]
```diff
@@ -280,4 +284,39 @@ public function getUnsavedDraftByUid(StripeSubscription $subscription): Subscrip
...
+    /**
+     * Ensures that a user with given email address is created if one doesn't already exist.
+     *
+     * @param StripeSubscription $subscription
+     * @param SubscriptionElement $subscriptionElement
+     * @return void
+     * @throws \yii\base\Exception
+     * @throws \yii\base\InvalidConfigException
+     */
+    private function maybeEnsureUser(StripeSubscription $subscription, SubscriptionElement $subscriptionElement): void
```
Comment (nfourtythree) [inline src/services/Subscriptions.php]: I would be tempted to have this as just `ensureUser()` and then wrap the call to it in a conditional. 

Can make things a little clearer in the fact that the method has "one job".

Not a biggie though so will let you decide on this one

File: src/services/Subscriptions.php [hunk 3]
```diff
@@ -280,4 +284,39 @@ public function getUnsavedDraftByUid(StripeSubscription $subscription): Subscrip
...
+    {
+        $plugin = Plugin::getInstance();
+        $settings = $plugin->getSettings();
+
+        if ($settings['createUserIfMissing']) {
```
Comment (nfourtythree) [inline src/services/Subscriptions.php]:
```suggestion
        if ($settings->createUserIfMissing) {
```

Might be worth flipping this to the inverse with an "exit early" strategy.

**Review by nfourtythree — CHANGES_REQUESTED** [review]

File: src/services/Subscriptions.php (hunk 2 above)
Comment (i-just) [reply src/services/Subscriptions.php]: That makes sense. Updated.

File: src/services/Subscriptions.php (hunk 3 above)
Comment (i-just) [reply src/services/Subscriptions.php]: Tweaked this, but the previous change means exiting early is no longer needed.

**Review by nfourtythree — APPROVED** [review]

**Pull request merged**